src/FSharp.Core/printf.fs

```diff
@@ -658,8 +658,28 @@ module internal PrintfImpl =
     
     /// Contains functions to handle left/right and no justification case for numbers
     module GenericNumber =
-
-        let isPositive (n: obj) = 
+        
+        let inline doubleIsPositive (n: double) =
+            n >= 0.0
+                // Ensure -0.0 is treated as negative (see https://github.com/dotnet/fsharp/issues/15557)
+                // and use bitwise comparison because floating point comparison treats +0.0 as equal to -0.0
+                && (BitConverter.DoubleToInt64Bits n <> BitConverter.DoubleToInt64Bits -0.0)
+        
+        let inline singleIsPositive (n: single) =
+            doubleIsPositive (float n)
```

Comment:
Related to the question of `NaN`, this has a chance of causing normalization and doing the "wrong thing". It should likely be explicitly handled instead, potentially using `single.IsPositive`